### PR TITLE
Add blog post about Sphinx ecosystem

### PR DESCRIPTION
**Pull request by astrojuanlu**
This post is more technical than we are used to, but I think it is very useful for our readers and users.

### CODE REVIEW

**Comment by astrojuanlu** [comment]
🤖 Rendered result: https://readthedocs-blog--150.org.readthedocs.build/sphinx-ecosystem-news-january-2022/

**Comment by astrojuanlu** [comment]
Since I quoted @choldgraf twice, pinging him in case he wants to have a look :)

File: sphinx-ecosystem-news-january-2022.rst [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+switcher <https://pydata-sphinx-theme.readthedocs.io/en/latest/user_guide/configuring.html#add-a-dropdown-to-switch-between-docs-versions>`_.
+As `Chris Holdgraf summarizes in this Twitter
+thread <https://twitter.com/choldgraf/status/1482435411301449729>`_,
+the new version brings other interesting additions,
+like `better depth control of the left
+sidebar <https://pydata-sphinx-theme.readthedocs.io/en/latest/user_guide/configuring.html#navigation-depth-and-collapsing-of-the-sidebar>`_
+and `custom SVG
+icons <https://pydata-sphinx-theme.readthedocs.io/en/latest/user_guide/configuring.html#local-image-icons>`_.
+
+Extensions
```
Comment (ericholscher) [inline sphinx-ecosystem-news-january-2022.rst:77]: These feel like good tweet content, and we reframe the blog post as annotated sphinx release notes?

**Review by ericholscher — COMMENTED** [review]
This is a great concept, unfortunately we don't have a plan to continue publishing something like this, so we need to reframe it as something that isn't the start of a series. I'd advocate for changing this to annotated Sphinx release notes, with a small section highlighting extensions or something. I feel like all of this would be good tweet content, especially the gif's, but I think we should also be focused on RTD some (do we support 4.4? How does a user enable it? Any caveats?)

I love the idea of being advocates for the Sphinx ecosystem, but we need a process in place before we start tracking all extensions and committing to writing ongoing content. I think Sphinx release notes feels like the right amount we can commit to, and something that is core to our users.

File: sphinx-ecosystem-news-january-2022.rst [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+----------
+
+Several new extensions saw new releases recently, including:
+
+`sphinx-codeautolink 0.9 <https://sphinx-codeautolink.readthedocs.io/en/stable/release_notes.html#id2>`_
+   sphinx-codeautolink is a Sphinx extension to automatically include links inside code blocks.
+   The 0.9 version now links Python builtin objects if intersphinx is properly configured
+   and has several logging improvements.
```
Comment (ericholscher) [inline sphinx-ecosystem-news-january-2022.rst:85]: We should be promoting this extension more! It seems super useful.

File: sphinx-ecosystem-news-january-2022.rst [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+`sphinxcontrib-constdata 1.0 <https://documatt.gitlab.io/sphinxcontrib-constdata/>`_
+   sphinxcontrib-constadata is a new Sphinx extension that allows documentation writers to
+   easily load data stored in CSV, JSON, and YAML files.
+   For example, it can be used to load UI labels (button labels, menu selection labels)
+   from external files instead of hardcoding them in the documentation for better maintenance,
+   for example:
+ 
+   .. code-block:: rst
+ 
+      Choose menu item :constdata:label:`menu.yaml?FileSaveAs`.
```
Comment (ericholscher) [inline sphinx-ecosystem-news-january-2022.rst:102]: Huh, interesting UX.

File: sphinx-ecosystem-news-january-2022.rst [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+The Executable Books Project team is working on several exciting things around MyST,
+including `the upcoming MyST-Parser 0.17 release <https://github.com/executablebooks/MyST-Parser/pull/507>`_
+and `direct integration with Jupyter notebooks <https://twitter.com/choldgraf/status/1485666900784730112>`_.
+
+.. figure:: /img/jupyter-myst.gif
+   :align: center
+   :alt: Preview of MyST integrated in Jupyter notebooks.
+
+   Preview of MyST integrated in Jupyter notebooks.
```
Comment (ericholscher) [inline sphinx-ecosystem-news-january-2022.rst:115]: This is an abrupt ending... We should close this out with something.

**Review by ericholscher — COMMENTED** [review]
A couple more content notes:

File: sphinx-ecosystem-news-january-2022.rst (hunk 2 above)
Comment (astrojuanlu) [reply sphinx-ecosystem-news-january-2022.rst:85]: Yep: https://github.com/humitos/sphinx-extensions/issues/18

File: sphinx-ecosystem-news-january-2022.rst (hunk 4 above)
Comment (astrojuanlu) [reply sphinx-ecosystem-news-january-2022.rst:115]: Fair enough 😅

File: sphinx-ecosystem-news-january-2022.rst (hunk 1 above)
Comment (astrojuanlu) [reply sphinx-ecosystem-news-january-2022.rst:77]: Not sure if you're advocating for removing this from the blog post and putting it in a Twitter thread, or just restructuring it a bit?

**Comment by astrojuanlu** [comment]
> I'd advocate for changing this to annotated Sphinx release notes, with a small section highlighting extensions or something.
> ...I think Sphinx release notes feels like the right amount we can commit to, and something that is core to our users.

Sounds good, I'll reframe the content and title a bit. Left a comment in the relevant thread.

> I think we should also be focused on RTD some (do we support 4.4? How does a user enable it? Any caveats?)

Good point, will add part of that too as well.

File: sphinx-ecosystem-news-january-2022.rst (hunk 1 above)
Comment (ericholscher) [reply sphinx-ecosystem-news-january-2022.rst:77]: Restructuring the post, but also using this content as a tweet. If we get better at tracking extensions, I think it's great twitter content, especially with the GIF's.

File: sphinx-ecosystem-news-january-2022.rst (hunk 2 above)
Comment (choldgraf) [reply sphinx-ecosystem-news-january-2022.rst:85]: cool - I hadn't heard about this before either!

**Review by choldgraf — COMMENTED** [review]
looks good to me in general - thanks for the shout-out!

**Comment by astrojuanlu** [comment]
Restructured the post a bit: changed the title to make it more focused on Sphinx 4.4, grouped all extensions and themes inside the same section, and added a paragraph on how to use this on Read the Docs.

**Review by ericholscher — APPROVED** [review]
Changes look good 👍

File: sphinx-4-4-release-other-ecosystem-news.rst [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+   This problem has now been fixed and autosummary properly points out the reason of the ``ImportError``.
+
+You can read the `complete Sphinx 4.4.0 release
+notes <https://www.sphinx-doc.org/en/master/changes.html#release-4-4-0-released-jan-17-2022>`_ online.
+
+Using Sphinx 4.4 on Read the Docs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Sphinx 4.4 is supported on Read the Docs. By default, RTD will install the latest version for you,
+as described in :ref:`our documentation <readthedocs:builds:external dependencies>`.
```
Comment (humitos) [inline sphinx-4-4-release-other-ecosystem-news.rst:67]: I'm happy this is very clear now in our documentation, it's easy to explain to user ("we install the latest version available"), and there are no surprises :100:. Way better than pinning to an old version and having to explain the "why" to users.

**Review by humitos — APPROVED** [review]
Very good blogpost! It would be awesome if we can sustain this over time!

**Comment by astrojuanlu** [comment]
Thanks both for the review! Merging and tweeting 💪🏽

**Pull request merged**